src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -93,6 +96,50 @@
   (instance? X509CRL x))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ValidX500Name
+  (schema/pred valid-x500-name?))
+
+(def Extension
+  "A map containing all the fields required to define an extension."
+  {:oid schema/Str
+   :critical (schema/pred #(not (nil? %)))
```

Comment:
Might be good to go further and assert that this can be coerced to a boolean value?